pkg/argocd/applications.go

```diff
@@ -202,3 +202,41 @@ func (builder *ApplicationBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithGitDetails applies git details to application definition.
+func (builder *ApplicationBuilder) WithGitDetails(gitRepo, gitBranch, gitPath string) *ApplicationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if gitRepo == "" {
+		glog.V(100).Infof("The 'gitRepo' of the argocd application is empty")
+
+		builder.errorMsg = "'gitRepo' parameter is empty"
+	}
+
+	if gitBranch == "" {
+		glog.V(100).Infof("The 'gitBranch' of the argocd application is empty")
+
+		builder.errorMsg = "'gitBranch' parameter is empty"
+	}
+
+	if gitPath == "" {
+		glog.V(100).Infof("The 'gitPath' of the argocd application is empty")
+
+		builder.errorMsg = "'gitPath' parameter is empty"
+	}
+
+	glog.V(100).Infof("Redefining argocd application %s in namespace: %s with git details",
```

Comment:
inconsistent use of colon, i.e.no colon after application and yes colon after namespace